### DISCUSSION TITLE
Device missing config entry check

### DISCUSSION THREAD

**Discussion by seang96** [widewing/ha-toyota-na#90] [General]
I am using a modified version to support my car from another brand so auth and client have additional patches for it to work properly, which is why I'm not putting this as an issue. I called the service to remote start and it worked this morning, but afterwards my requests went to missing config entry. I commented this section and it is working flawlessly again.

Can you provide any information on what the check does and what might be wrong to properly fix it?

For some context, my changes really only are to auth URL/key and then the client side is an additional header and I override the generic api method to replace the brand header with the proper one if its specified. All of the entities including the lock entity were working properly with this check enabled.

**Comment by dkmcgowan** [comment]
what file and line did you comment out?

**Reply by seang96** [reply]
Commenting out [__init__.py#L77-L79](https://github.com/widewing/ha-toyota-na/blob/master/custom_components%2Ftoyota_na%2F__init__.py#L77-L79) fixed my issue.

Thanks for this integration it is awesome!

**Comment by dkmcgowan** [comment]
That line was in there before me, looks like the assumption was only ever one config entry and it takes the first, so if yours worked you had one but must have had more than one.  Add a line to write to the log how many entries you have

**Reply by seang96** [reply]
Yeah it's showing a count of 2, I assume that is HASS number of devices on the integration? It only shows 1 device associated with the integration.

**Comment by seang96** [comment]
While I'm not sure what is causing multiple configs, perhaps a helper I have associated with the device? I updated it to loop through the devices because after restarting it for checking the count the first item is now the invalid one.

```
        # There is currently not a case with this integration where
        # the device will have more or less than one config entry
        if len(device.config_entries) == 0:
            _LOGGER.warning("Device missing config entry")
            return

        for entry_id in device.config_entries:
            if entry_id not in hass.data[DOMAIN]:
                _LOGGER.warning("Config entry not found")
                continue

            if "coordinator" not in hass.data[DOMAIN][entry_id]:
                _LOGGER.warning("Coordinator not found")
                continue

            coordinator = hass.data[DOMAIN][entry_id]["coordinator"]
            if coordinator.data is None:
                _LOGGER.warning("No coordinator data")

        if coordinator.data is None:
            _LOGGER.warning("No coordinator data")
            return
```

Is this something you'd be interested in a PR? I'd like to see what the other one is through.

**Comment by dkmcgowan** [comment]
Sure, toss up a PR and I will test as well, that code looks better for sure.  The only thing that would fail is if there were multiple valid configs and it picked the second one and you wanted the first, but not sure why that would happen.  This was my first HASS plugin to work on so I don't really know it that well, but either way this looks cleaner than what is there now.
